packages/compiler/formatter/print/printer.ts

```diff
@@ -774,16 +775,40 @@ export function printModelProperty(
       tryInline: true,
     }
   );
+  let id: Doc;
+  if (node.id.kind === SyntaxKind.StringLiteral && isStringSafeToUnquote(node.id, options)) {
+    id = node.id.value;
+  } else {
+    id = path.call(print, "id");
+  }
   return [
     multiline && isNotFirst ? hardline : "",
     decorators,
-    path.call(print, "id"),
+    id,
     node.optional ? "?: " : ": ",
     path.call(print, "value"),
     node.default ? [" = ", path.call(print, "default")] : "",
   ];
 }
 
+function isStringSafeToUnquote(id: StringLiteralNode, options: CadlPrettierOptions): boolean {
+  const unquotedRawText = getRawText(id, options).slice(1, -1);
+  if (id.value !== unquotedRawText) {
+    return false;
+  }
+  let hasError = false;
+  const scanner = createScanner(id.value, (d) => (hasError = true));
```

Comment:
@nguerrera triggered the scanner here to scan the string and see if it is just an identifier. Does this sounds like a good way to figure that out?

Reply:
Looks good.